### PR TITLE
Update DG for savepdf, some refractoring for savepdf stuffs, fix bugs

### LINKED ISSUE #184: Cannot delete customer even when there is no task associated with it
In the task lists, there is no task associated with customer#1, but the error shows this customer is allocated to a task.

<hr><sub>[original: Na-Nazhou/ped#19]<br/>
</sub>

### LINKED ISSUE #188: The app will show nothing after change the json file
After changing a name in json file into an empty string, restart the app, all driver, customer and task will breaks.

<hr><sub>[original: LuWenQ/ped#2]<br/>
</sub>

### PR DESCRIPTION
**Pull request by junnbang**
Fix #184 
new message when delete customer

image

Fix #188 
message alert when u open the app if user starts with empty manager.
image

### CODE REVIEW

File: src/main/java/seedu/address/model/task/TaskList.java [hunk 1]
```diff
@@ -113,20 +115,41 @@ public void setTask(Task taskToEdit, Task editedTask) {
         return tasksUnmodifiable;
     }
 
-    public List<Task> getSortedList(Comparator<Task> comparator) {
+    public static List<Task> getSortedList(List<Task> tasks, Comparator<Task> comparator) {
```
Comment (tysng) [inline src/main/java/seedu/address/model/task/TaskList.java]: This method is probably not needed, it's just `tasks.sort(comparator)`, don't need to use streams also.

File: src/main/java/seedu/address/model/task/TaskList.java [hunk 2]
```diff
@@ -113,20 +115,41 @@ public void setTask(Task taskToEdit, Task editedTask) {
...
+     * @param assignedTasks list of task that is assigned to drivers.
+     * @return driver list that contains only drivers with assigned tasks.
+     */
+    public static List<Driver> getDriversFromTasks(List<Task> assignedTasks) {
+        List<Driver> driverList = new ArrayList<>();
+        for (Task task : assignedTasks) {
+            assert task.getDriver().isPresent();
+
+            Driver driver = task.getDriver().get();
+            if (!driverList.contains(driver)) {
```
Comment (tysng) [inline src/main/java/seedu/address/model/task/TaskList.java]: `driverList` then can be a set and converted to list before you return

**Review by tysg — DISMISSED** [review]
Seems working

File: src/main/java/seedu/address/model/task/TaskList.java (hunk 1 above)
Comment (junnbang) [reply src/main/java/seedu/address/model/task/TaskList.java]: solved

File: src/main/java/seedu/address/model/task/TaskList.java (hunk 2 above)
Comment (junnbang) [reply src/main/java/seedu/address/model/task/TaskList.java]: good way!

**Review by tysg — APPROVED** [review]

**Pull request merged**